### PR TITLE
added option to manually select resourcetype

### PR DESCRIPTION
**Pull request by axfelix**
Hi Mark,

I've added the option to toggle simply picking a resourceType from a drop-down when assigning a DOI to an object, since Carla and Jessica preferred it that way, but I notice your code has diverged upstream from the version I was working from. Consider this just a stub for now; I'll clean up the commit later this week.

### CODE REVIEW

**Comment by mjordan** [comment]
Hi @axfelix, in the future, please open an issue before opening a PR, as described in the "Development and feedback" part of the README.

File: includes/utilities.inc [hunk 1]
```diff
@@ -16,15 +16,8 @@
...
   $dois = module_invoke_all('islandora_doi_framework_mint', $pid);
-  if (count($dois)) {
-    // We want the one minted in the last hook to fire.
-    $doi = array_pop($dois);
-  }
-  else {
-    return FALSE;
-  }
-  if (preg_match('#^10\..+/.+#', $doi)) {
-    return $doi;
```
Comment (mjordan) [inline includes/utilities.inc:27]: Why do you need to remove this regex validation?

File: includes/utilities.inc [hunk 2]
```diff
@@ -46,36 +39,29 @@ function islandora_doi_framework_persist_doi($doi, $pid) {
...
  */
-function islandora_doi_framework_update_doi($pid, $doi, $action = 'both') {
-  $updates = module_invoke_all('islandora_doi_framework_update', $pid, $doi, $action);
-  if (count($updates)) {
-    return TRUE;
-  }
-  else {
-    return FALSE;
-  }
+function islandora_doi_framework_update_doi($doi, $pid, $action = 'both') {
```
Comment (mjordan) [inline includes/utilities.inc:52]: Why are you switching the parameter order in this function?

File: includes/utilities.inc [hunk 3]
```diff
@@ -46,36 +39,29 @@ function islandora_doi_framework_persist_doi($doi, $pid) {
...
- *
- * @todo: Add DOIs and action for update functionality.
+ * Returns a list of PIDs or DOIs from a file.
  *
  * @param string $path
  *   The absolute path to the file.
  *
  * @return array
- *   A list of PIDs.
+ *   A list of PIDs or DOIs.
```
Comment (mjordan) [inline includes/utilities.inc:64]: Again, why this change?

File: islandora_doi_framework.module [hunk 4]
```diff
@@ -14,7 +14,7 @@ function islandora_doi_framework_menu() {
     'title' => 'DOI',
     'access arguments' => array('assign DOIs to Islandora objects'),
     'page callback' => 'drupal_get_form',
-    'page arguments' => array('islandora_doi_framework_manage_doi'),
+    'page arguments' => array('islandora_doi_framework_assign_doi'),
```
Comment (mjordan) [inline islandora_doi_framework.module:17]: The permission is 'manage', not 'assign'.

**Comment by mjordan** [comment]
@axfelix I can see the need to manually assign a resource type and I think adding it in the location you have is a good idea. But, one issue I will raise is that we need to accommodate batch DOI assignment via Drush. If we're going to provide a batch drush option, we need to provide a way for the module to automatically assign a resource type without user intervention. So, I'd like to see the (admittedly awkward) automated mapping take place all the time if invoked by drush, and keep your GUI option as you have implemented it.

However, there's a lot more in this PR than just the addition of the GUI option for assigning resource type . You say "I notice your code has diverged upstream from the version I was working from", but to me, it looks like you're changing the code to put the old version (the one you're working from) back. If my code changed, you should be using it, not opening a PR to put it back to match your old version. I added a few inline comments to the PR diff but then stopped when I realized that there were so many reversions.

Also, are you adding the DOI DDI module as a submodule of this repo? No problem if you want to, but because the DDI module is included in this PR without any explanation, I'm just assuming you are.

**Comment by mjordan** [comment]
This repo has a pull request template. Did it not show up when you opened the PR?

**Comment by axfelix** [comment]
Hi Mark,

I fear I've been misunderstood -- I really did mean "Consider this just a stub for now; I'll clean up the commit later this week."

I was doing the work to add this functionality in place because it was what we wanted and figured I'd add a toggle and reconcile changes with your code when I was done so it'd get upstream; I didn't notice it had diverged so much until opening this PR, but I thought I'd leave it open so neither of us forgot about it and clean up the PR later. How would you like me to proceed?

**Comment by mjordan** [comment]
Sorry if I misunderstood. How about this:

* Keep this PR open as a placeholder, but we won't merge it; we'll just close it when all the work is done in separate PRs.
* Open an issue and corresponding PR for each major set of changes you want to make, e.g., integrate the resourceType UI, add the DOI DDI module, whatever else you want to add. The reason I'd like to have an issue first is so we can discuss possible implementations before staring work.
* Name each branch where you're doing the work afte the issue number, e.g., 'issue-2'.
* Please use the PR template.

Sorry to be so anal about workflow here. In the absence of automated tests I'd rather chunk the work out into discrete PRs so we can test more reliably as we go.

**Comment by axfelix** [comment]
Sounds good! Thanks.

**Comment by mjordan** [comment]
@axfelix anything in this original stub PR that hasn't been addressed elsewhere? If so, let me know; if not, please close.
